src/app/modules/group/components/pending-request/pending-request.component.ts

```diff
@@ -49,15 +51,78 @@ export class PendingRequestComponent implements OnInit, OnChanges {
   currentSort: string[] = [];
   includeSubgroup = false;
   collapsed = true;
-  status: 'loading' | 'loaded' | 'empty' |'error' = 'loading';
+
+  state: 'fetching' | 'ready' |'error' = 'fetching';
 
   ongoingActivity: Activity = 'none';
 
+  private dataFetching = new Subject<{ groupId?: string, includeSubgroup: boolean, sort: string[] }>();
+
+  private dataUpdating = new Subject<{ action: Action, requests: PendingRequest[] }>();
```

Comment:
When reading it the term `dataUpdating` is not clear (looks like it is the same as data fetching).
...
Now I've read what we do with them:
It is quite messy. Actually, your two subjects update the same state, so they cannot really be separated as they may conflict. (more comments below)

Reply:
fixed, I removed it